browser/scripts/ui/xhrForm.js

```diff
@@ -0,0 +1,288 @@
+if (typeof VizorUI === 'undefined')
+	var VizorUI = {}
+
+if (typeof uiEvent === 'undefined')
+	uiEvent = {}
+
+uiEvent.xhrFormSuccess	= 'xhrFormSuccess'		//	dispatched on document
+uiEvent.xhrFormUIReset	= 'xhrFormUIReset'		//	dispatched on form
+
+/**
+ * enables a form to submit and process its response via xhr call
+ * validation is done in backend
+ * @param $form
+ * @param onSuccess
+ */
+VizorUI.setupXHRForm = function($form, onSuccess) {	// see views/account/signup for example
+
+	var xhrEnabled = $form.data('xhrEnabled')
+	if (xhrEnabled) {
+		msg("ERROR: setupXHRForm but form already enabled " + $form.attr('id'))
+		return false
+	}
+
+	var hasFiles = $form.length && ($form[0].querySelectorAll('input[type="file"]').length > 0)
+
+	if (typeof onSuccess !== 'function') {
+		onSuccess = function(response) {
+			if (response && (typeof response.redirect !== 'undefined') && (response.redirect)) {
+				window.location.href = response.redirect
+				return
+			}
+			// append a message to form
+			var message = jQuery('<p></p>')
+							.text(response.message)
+							.addClass('xhr success message')
+			$form.append(message)
+			$form.find(':focus').trigger('blur')
+
+			var detail = {
+				id: 		$form.attr('id'),
+				response: 	response
+			}
+			var successEvent = new CustomEvent(uiEvent.xhrFormSuccess, {detail: detail});
```

Comment:
could also use a Flux action/dispatch here, but I suppose this has the same effect.